File: frontend/src/entidades/Plato.ts
import IngredienteCantidad from "./IngredienteCantidad";

 class Plato{
    id=0;
    imagenPath = "";
    nombre = "";
    precio=0;
    rubro = "";
    ingredientes:IngredienteCantidad[] = [];
    urlImagenPath = "";
}

export default Plato

File: backend/src/main/java/com/tpvue/backend/entities/Plato.java
package com.tpvue.backend.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;


@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Plato {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String imagenPath;
    private String nombre;
    private double precio;
    private String rubro;

    @OneToMany(fetch = FetchType.LAZY)
    private Set<IngredienteCantidad> ingredientes;

    private String urlImagenPath;

}
